src/autolabel/labeler.py

```diff
@@ -478,8 +483,15 @@ def generate_explanations(
 
         return seed_examples
 
-    def clear_cache(self):
-        if self.cache:
-            self.cache.clear()
-        else:
-            logger.error("No cache to clear")
+    def clear_cache(self, use_ttl: bool = True):
+        """
+        Clears the generation and transformation cache from autolabel.
+        Args:
+            use_ttl: If true, only clears the cache if the ttl has expired.
+            If this is true, only the transform cache will be cleared (as the generation cache does not use ttl)
+        """
+        if self.generation_cache and not use_ttl:
```

Comment:
should we make the default ttl to be a large number so we don't need to handle it any different than the transform cache?

Reply:
makes sense, done